asyncio_mqtt/client.py

```diff
@@ -889,6 +895,8 @@ async def __aexit__(
                 " disconnection."
             )
             await self.force_disconnect()
+        finally:
+            self._lock.release()
```

Comment:
Good use of `finally` 👍

Reply:
Hmm, there are some subtle race conditions in this __aexit__ method now. E.g., if `disconnect` raises, we never call `force_disconnect`. That is bad!

My suggestion:
 * **Breaking change**: Make `force_disconnect` private (call it `_force_disconnect`).
 * Make `_force_disconnect` sync (there is no reason for it to be async anyhow).
 * Move `_force_disconnect` into the `finally` block (just above the lock release)

That should fix the race conditions. Note that `_force_disconnect`: (1) never raises and (2) is a no-op if already disconnected.

Does it make sense?

Reply:
Mention the breaking change in the changelog as well. In fact, make a changelog entry about this new reentrant Client feature! 😄 Our users would want to know about it!